nilearn/input_data/tests/test_masker_validation.py

```diff
@@ -0,0 +1,65 @@
+import nibabel
+import numpy as np
+
+from sklearn.externals.joblib import Memory
+
+from nilearn.input_data.masker_validation import check_embedded_nifti_masker
+from nilearn.input_data import MultiNiftiMasker, NiftiMasker
+
+
+class C:
+
+    def __init__(self, mask=None, smoothing_fwhm=None,
+                 standardize=False, detrend=False,
+                 low_pass=None, high_pass=None, t_r=None,
+                 target_affine=None, target_shape=None,
+                 mask_strategy='background', mask_args=None,
+                 memory=Memory(cachedir=None), memory_level=0,
+                 n_jobs=1, verbose=0,
+                 dummy=None):
+        self.mask = mask
+
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.detrend = detrend
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.mask_strategy = mask_strategy
+        self.mask_args = mask_args
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.verbose = verbose
+        self.dummy = dummy
+
+
+def test_check_embedded_nifti_masker():
+    c = C()
+    masker = check_embedded_nifti_masker(c)
+    assert(isinstance(masker, MultiNiftiMasker))
```

Comment:
You probably want to check strict type equality rather than inheritance: `type(masker) == MultiNiftiMasker`